neuro_san/internals/chat/data_driven_chat_session.py

```diff
@@ -105,6 +108,14 @@ async def set_up(self, invocation_context: InvocationContext,
         # Reset what we might have created before.
         await self.delete_resources()
 
+        # Update sly data, if any.
+        # Note that since this instance is the owner of the sly_data,
+        # any update here should get transmitted to all the other graph components
+        # because it is expected they share the reference and only interact with it
+        # in a read-only fashion.
```

Comment:
I think the other components can update `sly_data`, can't they?  I understand this piece of code just moved up - just wondering.

Reply:
Yes. This is simply moving when the single sly_data instance for the entire request gets initialized earlier.